### DISCUSSION TITLE
Drawing on PDF

### DISCUSSION THREAD

**Discussion by usagibear** [wojtekmaj/react-pdf#1421] [Q&A]
What are some ideas for implementing drawing on PDF files, say to mark fixes or changes, similar to what Firefox implements in the default PDF.js implemented PDF viewer? Or how Edge implements PDF annotations? Any ideas or thought would be of great help. Thanks!

Firefox
![image](https://user-images.githubusercontent.com/114092778/229250057-1e79b58c-7fc3-4389-8e0d-0ac1dfa495ce.png)

Edge
![image](https://user-images.githubusercontent.com/114092778/229250117-08fcc025-dbc0-4488-8877-e981285bf86f.png)

**Comment by wojtekmaj** [comment]
While not exact thing you're looking for, check out this example on how to add text on canvas:

https://codesandbox.io/s/react-pdf-watermark-5wylx

I guess drawing shapes wouldn't be very different 🤔
